semtransforms/transformations/strong.py

```diff
@@ -67,15 +92,26 @@ def if0error(finder: FindStatements, parents: List[Node], stmts: Content, contex
     return lambda: stmts.replace(If(Constant("int", "0"), verifier.error_call(), None))
 
 
+
 @find_statements(length=1, modifiable_length=False, context=True)
 def to_method(finder: FindStatements, parents: List[Node], stmts: Content, context: ContextVisitor):
     if isinstance(stmts[0], Compound) and not\
             (finder.has_break(stmts[0]) or finder.has_jumps(stmts[0]) or finder.has_return(stmts[0])):
-        name = context.free_name()
+
+        # Handling struct ref does not seem to be really supported
+        # TODO: Fix and drop this condition
+        if finder.has_struct_ref(stmts[0]): return
+        
+        if len(stmts[0].block_items) == 0: return
+
+        name = context.free_name(prefix = "func_")
         # get names and types of all variables defined before used in this block
         urs = unknown_references(stmts[0])
-        names = {ur[0][0].name for ur in urs}
-        original_params = [context.value(id) for id in names]
+        local_urs = [ur for ur in urs if ur[0][0].name not in context.globals]
+        local_names = {ur[0][0].name for ur in local_urs}
+        original_params = [context.value(id) for id in local_names]
+
+        # Why is this needed? We can process variable array sizes via pointers
```

Comment:
you can do sizeof on an array, but not on a pointer.
So you would have to look through all usages and calculate and transfer the size if it is used